contracts/YieldLeverBase.sol

```diff
@@ -12,20 +12,48 @@ import "@yield-protocol/utils-v2/contracts/token/IERC20.sol";
 import "@yield-protocol/utils-v2/contracts/token/TransferHelper.sol";
 import "@yield-protocol/vault-v2/utils/Giver.sol";
 import "@yield-protocol/vault-v2/FlashJoin.sol";
+import "@yield-protocol/vault-v2/Join.sol";
+import "@yield-protocol/utils-v2/contracts/interfaces/IWETH9.sol";
 
 error FlashLoanFailure();
 error SlippageFailure();
 
-contract YieldLeverBase is IERC3156FlashBorrower {
+interface Ladle is ILadle {
+    function give(bytes12 vaultId_, address receiver)
+        external
+        payable
+        returns (DataTypes.Vault memory vault);
+}
+
+abstract contract YieldLeverBase is IERC3156FlashBorrower {
+    using TransferHelper for IWETH9;
+    using TransferHelper for IERC20;
+
     /// @notice The Yield Ladle, the primary entry point for most high-level
     ///     operations.
-    ILadle public constant ladle =
-        ILadle(0x6cB18fF2A33e981D1e38A663Ca056c0a5265066A);
+    Ladle public constant LADLE =
+        Ladle(0x6cB18fF2A33e981D1e38A663Ca056c0a5265066A);
+
     /// @notice The Yield Cauldron, handles debt and collateral balances.
-    ICauldron public constant cauldron =
+    ICauldron public constant CAULDRON =
         ICauldron(0xc88191F8cb8e6D4a668B047c1C8503432c3Ca867);
 
+    /// @notice WEth.
+    IWETH9 public constant WETH =
+        IWETH9(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2);
+
+    bytes6 constant ASSET_ID_MASK = 0xFFFF00000000;
+
+    // TODO: Events?
+    event LeveredUp();
+    event Repaid();
+    event Closed();
+
     /// @notice The operation to execute in the flash loan.
+    ///
+    ///     - BORROW: Invest
+    ///     - REPAY: Unwind before maturity
+    ///     - CLOSE: Unwind after maturity
     enum Operation {
```

Comment:
The `enum` should also go before any state variables, I think between the `using` statements and the events.

Please also put in consecutive lines the non-contract constants (`FLASH_LOAN_RETURN` and `ASSET_ID_MASK`). Same for the contract constants (`WETH`, `Cauldron`, `Ladle`, `Giver`, in this order)